curator-examples/src/main/java/discovery/DiscoveryExample.java

```diff
@@ -192,6 +194,14 @@ private static void listInstances(ServiceDiscovery<InstanceDetails> serviceDisco
                 }
             }
         }
+        catch ( Exception e )
+        {
+            if ( e instanceof KeeperException.NoNodeException )
+            {
+                System.err.println("There are no registered instances.");
+                return;
+            }
```

Comment:
any other exception should be rethrown

Reply:
Yes, that's what I thought, and I hope what I'm doing now is the same as what we discussed.